docs/tutorials/articles/sales_dashboard/step_02/step_02.md

```diff
@@ -0,0 +1,198 @@
+```
+
+[Part](../../../../refmans/gui/viselements/generic/part.md) allows you to group and style visual elements together.
+Here, the [container](../../../../userman/gui/styling/stylekit.md) class is a predefined style
+that will limit the width of visual elements contained in the part.
+
+[Text](../../../../refmans/gui/viselements/generic/text.md) is a simple text visual element. Here we use the
+Markdown (md) mode to display a title and make the word "State". We can also color bold text in orange
+by using CSS. Create a new CSS file with the same name as the Python file and add the following code:
+
+```css
+strong,
+b {
+    font-weight: bold;
+    color: var(--color-primary);
+}
+```
+
+Let's now add a new container for the filters:
+
+```python
+        with tgb.part(class_name="card"):
+            with tgb.layout(columns="1 2 1"):
+                with tgb.part():
+                    tgb.text("Filter **From**", mode="md")
+                with tgb.part():
+                    tgb.text("Filter Product **Category**", mode="md")
+                with tgb.part(class_name="text-center"):
+                    tgb.button(
+                        "Apply",
+                    )
+```
+
+[Card](../../../../userman/gui/styling/stylekit.md) is a predefined style that will regroup visual elements in
+a white box. [layout](../../../../refmans/gui/viselements/generic/layout.md) allows you to create columns
+for visual elements. Here we create 3 columns with a ratio of 1:2:1, the second column will be twice as wide as the
+first and last columns. The contents of each column then needs to be separated in parts.
+
+![Layout](images/layout.png){ width=90% : .tp-image-border }
+
+We can now add [date selectors](../../../../refmans/gui/viselements/generic/date.md), 
+[selectors](../../../../refmans/gui/viselements/generic/selector.md) and a 
+[button](../../../../refmans/gui/viselements/generic/button.md) to apply the filters:
+
+```python
+        with tgb.part(class_name="card"):
+            with tgb.layout(columns="1 2 1"):
+                with tgb.part():
+                    tgb.text("Filter **From**", mode="md")
+                    tgb.date("{start_date}")
+                    tgb.text("To")
+                    tgb.date("{end_date}")
+                with tgb.part():
+                    tgb.text("Filter Product **Category**", mode="md")
+                    tgb.selector(
+                        value="{selected_category}",
+                        lov="{categories}",
+                        on_change=change_category,
+                        dropdown=True,
+                    )
+                    tgb.text("Filter Product **Subcategory**", mode="md")
+                    tgb.selector(
+                        value="{selected_subcategory}",
+                        lov="{subcategories}",
+                        dropdown=True,
+                    )
+                with tgb.part(class_name="text-center"):
```

Comment:
Why do you need a part if there's only one element?

Reply:
Parts are a good way to visually separate in the code the contents of different columns. Also here I use part to center the button horizontally. Is there another way to center the button without using part nor CSS?